Framework/DataHandling/inc/MantidDataHandling/LoadMuonNexusV2Helper.h

```diff
@@ -6,13 +6,21 @@
 // SPDX - License - Identifier: GPL - 3.0 +
 #pragma once
 #include "MantidAPI/WorkspaceGroup_fwd.h"
+#include "MantidAPI/Workspace_fwd.h"
+#include "MantidDataObjects/Workspace2D_fwd.h"
 #include "MantidGeometry/IDTypes.h"
 #include "MantidNexus/NexusClasses.h"
 
 #include <vector>
 
 namespace Mantid {
 namespace DataHandling {
+namespace MuonNexus {
+struct SampleInformation {
+  double magneticField;
+  double temperature;
+};
+} // namespace MuonNexus
 namespace LoadMuonNexusV2Helper {
```

Comment:
Every free func which takes a NXEntry should be bundled into a class which takes that as a constructor.
This way you can also create some private implementations...etc. or reuse the same NXEntry tag to do a few things